src/v/redpanda/BUILD

```diff
@@ -95,11 +96,33 @@ redpanda_cc_library(
     ],
 )
 
+bool_flag(
+    name = "lto",
+    build_setting_default = True,
```

Comment:
Should we turn this on only in release mode? Which means default to False and set to true only in `--config=release`?

Reply:
It doesn't make any difference but yes we should probably only add it in release.

Reply:
Hmm I am now wondering what the best way to go about this is.

I see in the openssl build for example we create an extra config flag for this? Is that the right way?

Another possibility would be to default the lto flag to False and set it to true from the .bazelrc release config. From testing however this means it can't be unset on the cli anymore.

Reply:
> Another possibility would be to default the lto flag to False and set it to true from the .bazelrc release config.

This is what I was imagining.

`--config=` is just a shorthand for setting flags and for boolean flag the last value wins. The following **should** work in terms of being a release build and not having lto (if not it's a bug and I will file in Bazel):

```
bazel build --config=release --//src/v/redpanda:lto=false //:redpanda
```

Reply:
Oh so the order matters. I had `--config` last. Let me try.

Reply:
Ok yes great it works like that. Pushed like that now.